skfp/distances/tanimoto.py

```diff
@@ -0,0 +1,332 @@
+    else:
+        raise TypeError(
+            f"Both vec_a and vec_b must be of the same type, either numpy.ndarray "
+            f"or scipy.sparse.csr_array, got {type(vec_a)} and {type(vec_b)}"
+        )
+
+
+@validate_params(
+    {
+        "vec_a": ["array-like", csr_array],
+        "vec_b": ["array-like", csr_array],
+    },
+    prefer_skip_nested_validation=True,
+)
+def tanimoto_binary_distance(
+    vec_a: Union[np.ndarray, csr_array], vec_b: Union[np.ndarray, csr_array]
+) -> float:
+    """
+    Tanimoto distance for vectors of binary values.
+
+    Computes the Tanimoto distance for binary data between two input arrays
+    or sparse matrices by subtracting the similarity from 1, using to
+    the formula:
+
+    .. math::
+
+        dist(vec_a, vec_b) = 1 - sim(vec_a, vec_b)
+
+    The calculated distance falls within the range `[0, 1]`.
+    Passing all-zero vectors to this function results in a distance of 0.
+
+    Parameters
+    ----------
+    vec_a : {ndarray, sparse matrix}
+        First binary input array or sparse matrix.
+
+    vec_b : {ndarray, sparse matrix}
+        Second binary input array or sparse matrix.
+
+    Returns
+    ----------
+    distance : float
+        Tanimoto distance between vec_a and vec_b.
+
+    Examples
+    ----------
+    >>> from skfp.distances import tanimoto_binary_distance
+    >>> import numpy as np
+    >>> vec_a = np.array([1, 0, 1])
+    >>> vec_b = np.array([1, 0, 1])
+    >>> dist = tanimoto_binary_distance(vec_a, vec_b)
+    >>> dist
+    0.0
+
+    >>> from skfp.distances import tanimoto_binary_distance
+    >>> from scipy.sparse import csr_array
+    >>> vec_a = csr_array([[1, 0, 1]])
+    >>> vec_b = csr_array([[1, 0, 1]])
+    >>> dist = tanimoto_binary_distance(vec_a, vec_b)
+    >>> dist
+    0.0
+    """
+
+    return 1 - tanimoto_binary_similarity(vec_a, vec_b)
+
+
+@validate_params(
+    {
+        "vec_a": ["array-like", csr_array],
+        "vec_b": ["array-like", csr_array],
+    },
+    prefer_skip_nested_validation=True,
+)
+def tanimoto_count_similarity(
+    vec_a: Union[np.ndarray, csr_array], vec_b: Union[np.ndarray, csr_array]
+) -> float:
+    """
+    Tanimoto simlarity for vectors of count values.
+
+    Computes the Tanimoto similarity [1]_ for count data between two input arrays
+    or sparse matrices using the formula:
+
+    .. math::
+
+        sim(vec_a, vec_b) = \\frac{vec_a \\cdot vec_b}{\\|vec_a\\|^2 + \\|vec_b\\|^2 - vec_a \\cdot vec_b}
+
+    Calculated similarity falls within the range of `[0, 1]`.
+    Passing all-zero vectors to this function results in similarity of 1.
+
+    Note that Numpy version is optimized with Numba JIT compiler, resulting in significantly faster
+    performance compared to SciPy sparse arrays. First usage may be slightly slower due to Numba compilation.
+
+    Parameters
+    ----------
+    vec_a : {ndarray, sparse matrix}
+        First binary input array or sparse matrix.
```

Comment:
Here we are in body of count tanimoto... Shouldn't it be count for instead of binary input?

Reply:
+1